### PR TITLE
Remove English tab from translations UI

### PR DESCRIPTION
**Pull request by clouser2**
### Description

All English translations are contained in other parts of the UI and are duplicative and confusing. Additionally, maintaining multiple updaters of this text requires duplicating any validation logic as well, which is particularly problematic for program statuses (since there is no internal/external distinction between the status display name).

Now, if one navigates to the English translations UI (or any unsupported locale), they're redirected to the program/question index page with an error toast message. Note that previously we rendered the translation UI for ANY locale that was parseable from the URL.

There are a few views that generate a link to the translation UI. Previously, they were forming a link to the "English" translation form. Instead of adding logic to each of these for choosing the first locale, I introduced a `/translations` route that performs the "choose the first non-English locale" logic and had calling views link to that.

Finally, left links back to the program/question editing views so that it was easier to edit the English text.

## Screenshots

Program translations:
Screen Shot 2022-08-05 at 11 38 58 AM

Question translations:
Screen Shot 2022-08-05 at 11 38 49 AM

## Release notes:

Removing the English translations tab since the information is already contained in the program / question builders.

### Checklist

- [X] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

#2754

### CODE REVIEW

File: server/app/controllers/admin/AdminQuestionTranslationsController.java [hunk 1]
```diff
@@ -30,17 +35,41 @@ public class AdminQuestionTranslationsController extends CiviFormController {
...
+   * Redirects to the first non-English locale eligible for translations for a a program.
+   *
+   * @param request the current {@link Http.Request}
+   * @param id the ID of the program to update
+   */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result redirectToFirstLocale(Http.Request request, long id) {
+    if (maybeFirstLocaleForTranslations.isEmpty()) {
+      return redirect(routes.AdminQuestionController.index().url())
+          .flashing("error", "Translations are not enabled for this configuration");
```
Comment (shanemc-goog) [inline server/app/controllers/admin/AdminQuestionTranslationsController.java:67]: Question: should toast messages be sentence structure?  EG there isn't punctuation here.

Comment (shanemc-goog) [inline server/app/controllers/admin/AdminQuestionTranslationsController.java:67]: A similar message below is "Unsupported locale: %s", should these be the same or similar structures?

**Review by shanemc-goog — CHANGES_REQUESTED** [review]
just nits

File: server/app/controllers/admin/AdminQuestionTranslationsController.java (hunk 1 above)
Comment (clouser2) [reply server/app/controllers/admin/AdminQuestionTranslationsController.java:66]: Fair. I updated to:

```
The %s locale is not supported
```

Hopefully these error messages will never be seen by a user :).

File: server/app/services/TranslationLocales.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Function;
+import play.i18n.Lang;
+import play.i18n.Langs;
+
+/** Helpers to support locales in which translations are supported. */
+public class TranslationLocales {
+
+  private final ImmutableList<Locale> localesForTranslation;
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/services/TranslationLocales.java]: Good suggestion. Done.

File: server/app/services/TranslationLocales.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  private final ImmutableMap<String, Locale> languageTagToLocale;
+
+  @Inject
+  public TranslationLocales(Langs langs) {
+    this.localesForTranslation =
+        langs.availables().stream()
+            .map(Lang::toLocale)
+            .filter(locale -> !LocalizedStrings.DEFAULT_LOCALE.equals(locale))
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/services/TranslationLocales.java]: Thanks! Done.

File: server/app/services/TranslationLocales.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            .collect(ImmutableList.toImmutableList());
+    this.languageTagToLocale =
+        this.localesForTranslation.stream()
+            .collect(ImmutableMap.toImmutableMap(Locale::toLanguageTag, Function.identity()));
+  }
+
+  /** Returns whether the provided locale is supported for translations. */
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/services/TranslationLocales.java]: Took a stab at it and renamed to `fromLanguageTag`.

File: server/app/services/TranslationLocales.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  public Optional<Locale> getSupportedLocale(String languageTag) {
+    return Optional.ofNullable(languageTagToLocale.get(languageTag));
+  }
+
+  /**
+   * Holds a list of {@link java.util.Locale} objects representing the non-English locales used for
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/services/TranslationLocales.java]: Fixed.

File: server/conf/routes [hunk 6]
```diff
@@ -33,6 +33,7 @@ GET     /admin/programs/:programId/statuses controllers.admin.AdminProgramStatus
 POST    /admin/programs/:programId/statuses/delete controllers.admin.AdminProgramStatusesController.delete(request: Request, programId: Long)
 
 # Routes for managing program localization
+GET     /admin/programs/:programId/translations controllers.admin.AdminProgramTranslationsController.redirectToFirstLocale(request: Request, programId: Long)
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/conf/routes]: Sure. Made each of them `translations/edit`

**Review by clouser2 — COMMENTED** [review]
Feedback incorporated!

File: server/app/controllers/admin/AdminProgramTranslationsController.java [hunk 7]
```diff
@@ -24,13 +27,39 @@ public class AdminProgramTranslationsController extends CiviFormController {
...
+    this.service = checkNotNull(service);
+    this.translationView = checkNotNull(translationView);
+    this.formFactory = checkNotNull(formFactory);
+    this.translationLocales = checkNotNull(translationLocales);
+    this.maybeFirstLocaleForTranslations =
+        this.translationLocales.localesForTranslation().stream().findFirst();
+  }
+
+  /**
+   * Redirects to the first non-English locale eligible for translations for a a program.
```
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/controllers/admin/AdminProgramTranslationsController.java]: Done, here and in the question translation controller as well.

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**